utils/frame/benchmarking-cli/src/writer.rs

```diff
@@ -45,11 +47,6 @@ pub fn write_trait(file: &mut File, batches: Vec<BenchmarkBatch>) -> Result<(),
 
 		// only create new trait definitions when we go to a new pallet
 		if batch.pallet != current_pallet {
-			if !current_pallet.is_empty() {
-				// close trait
-				write!(file, "}}\n").unwrap();
-			}
```

Comment:
I guess this is because generating multiple WeightInfo is no longer supported ?

Reply:
This is now moved to the bottom of the code. I had to move it because at the top of the loop, we are already working with a new file if the file has changed. Instead we `peek` at the bottom, close out the file if we know we will change files next.

Reply:
is this true for write_trait ? for what I see the usage is:
```rust
						let mut file = crate::writer::open_file("traits.rs")?;
						crate::writer::write_trait(&mut file, batches.clone())?;
```
thus traits.rs hold all trait definition in one file, if batches contains multiple pallet then their trait definition are not ending

Reply:
yeah, I didnt update write_trait... let me look at that next

Reply:
Actually I think I wont update. right now it creates a single file with all the traits, and this is honestly most convenient. This will not be used in the auto-generation pipeline, but is just a helper for manual work needed to be written in the palllet.